### PR TITLE
Timing fix

### PR DESCRIPTION
**Pull request by psppalash**
Adding sync.Waitgroup to wait for greaselib to start.
Moving greasego startup related code into lambda(callback function) that is passed onto greaselib.
uv_async_send(&asyncInternalCommand); was being called before asyncInternalCommand was getting initialized. (race condition) This was fixed by assuring that asyncInternalCommand is intialized first.

### CODE REVIEW

**Comment by psppalash** [comment]
Need to rebase.

File: logconfig/manager.go [hunk 1]
```diff
@@ -450,137 +450,141 @@ func InitLogManager(config *maestroConfig.YAMLMaestroConfig) (err error) {
...
-			// 		fmt.Printf("ERROR in toCloud target CB %s\n", err.Str)
-			// 	} else {
-			// 		buf := data.GetBufferAsSlice()
-			// 		DEBUG(s := string(buf))
-			// 		debugging.DEBUG_OUT("CALLBACK %+v ---->%s<----\n\n",data,s);
-			// 		client.SubmitLogs(data,buf)
-			// 	}
-			// }
-		}
+			debugging.DEBUG_OUT("%+v\n", opts.FileOpts)
```
Comment (mray190) [inline logconfig/manager.go:507]: Will these debug statements need to be removed? Looks like they were originally commented out?

Comment (costanic) [reply logconfig/manager.go:507]: I think this was originally on line 506, which seems to still be active.  I think the maestro DEBUG flag ulitimately controls whether this is enabled?  not sure.

Comment (psppalash) [reply logconfig/manager.go:507]: Yes, I think DEBUG flag controls if these statements are enabled.

**Pull request merged**